### PR TITLE
ENH: Use skyfield to propagate and convert

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Added an instrument with skyfield support.

Skyfield runs on top of sgp4 and handles the conversions.  This sets up a test instrument that can either be a replacement for `missions_sgp4` or an additional instrument.

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

```
import datetime as dt
import numpy as np
import pandas as pds
import pysat

from pysatMissions.instruments import missions_sgp4
from pysatMissions.instruments import missions_skyfield

# Use SPORT TLEs as an example
line1 = '1 55129U 98067UW  23039.82653086  .00049915  00000-0  78305-3 0  9991'
line2 = '2 55129  51.6397 243.8305 0001856 340.7022  19.3898 15.53177014  6499'


file_date_range = pds.date_range(dt.datetime(2022, 12, 30),
                                 dt.datetime(2023, 2, 28))

old_sat = pysat.Instrument(inst_module=missions_sgp4, tle1=line1, tle2=line2,
                           use_header=True, file_date_range=file_date_range)
new_sat = pysat.Instrument(inst_module=missions_skyfield, tle1=line1, tle2=line2,
                           use_header=True, file_date_range=file_date_range)

old_sat.load(2023, 39)
new_sat.load(2023, 39)

keys = ['position_eci_x', 'position_eci_y', 'position_eci_z',
        'velocity_eci_x', 'velocity_eci_y', 'velocity_eci_z',
        'position_ecef_x', 'position_ecef_y', 'position_ecef_z',
        'velocity_ecef_x', 'velocity_ecef_y', 'velocity_ecef_z',
        'geod_latitude', 'geod_longitude', 'geod_altitude']

print("{:25s} {:10s} {:10s} {:10s}".format('key', 'min', 'max', 'std'))
print("{:25s} {:10s} {:10s} {:10s}".format('---', '---', '---', '---'))
for key in keys:
    delta = old_sat[key] - new_sat[key]
    print("{:20} {:10.5f}  {:10.5f}  {:10.5f}".format(key, np.min(delta),
                                                      np.max(delta),
                                                      np.std(delta)))

```
yields 
```
key                       min        max        std       
---                       ---        ---        ---       
position_eci_x        -32.63257    32.58374    22.57192
position_eci_y        -24.96048    24.96973    17.29847
position_eci_z        -10.91105    10.91512     7.56270
velocity_eci_x         -0.03690     0.03679     0.02555
velocity_eci_y         -0.02821     0.02821     0.01956
velocity_eci_z         -0.01233     0.01233     0.00855
position_ecef_x        -0.00298     0.00296     0.00175
position_ecef_y        -0.00296     0.00298     0.00174
position_ecef_z        -0.00000     0.00000     0.00000
velocity_ecef_x        -0.49480     0.49140     0.29159
velocity_ecef_y        -0.49190     0.49494     0.29047
velocity_ecef_z        -0.00000     0.00000     0.00000
geod_latitude          -0.00000     0.00000     0.00000
geod_longitude          0.00002     0.00003     0.00000
geod_altitude          -0.00000     0.00001     0.00000
```

**Test Configuration**:
* Mac 12.6.3
* python 3.10.0

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by jklenzing** [comment]
NOTE: need to document different ECI coordinate systems for skyfield and sgp4.

**Comment by rstoneback** [comment]
I'm guessing using WGS72 doesn't help with the difference in ECI? Does it impact the orbits otherwise?

Looks like we need a third source of TLE to ECI conversions to break the tie 😜 

Good news is the positions in ECEF and LLA are nearly the same.

**Comment by jklenzing** [comment]
The problem is that there are multiple "ECI" coords, so we need to label these more specifically

File: pysatMissions/instruments/missions_skyfield.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+        ind = times <= (2 * np.pi / mean_motion * 60)
+        times = times[ind]
+        index = index[ind]
+
+    skytimes = skyapi.load.timescale().from_datetimes(index.tz_localize('UTC'))
+
+    esat = skyapi.EarthSatellite.from_satrec(satrec, skyapi.load.timescale())
+    # Generate object with info for all times
+    sat_obj = esat.at(skytimes)
+
```
Comment (JonathonMSmith) [inline pysatMissions/instruments/missions_skyfield.py:216]: This is where we should put that this is the TEME ECI coords?
Skyfield docs say True Equator and Equinox of date, is that the same as true equator and mean equinox?

File: pysatMissions/instruments/missions_skyfield.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    meta['Epoch'] = {
+        meta.labels.units: 'Milliseconds since 1970-1-1',
+        meta.labels.notes: 'UTC time at middle of geophysical measurement.',
+        meta.labels.desc: 'UTC seconds',
+        meta.labels.name: 'Time index in milliseconds'}
+    for v in ['x', 'y', 'z']:
+        meta['position_eci_{:}'.format(v)] = {
+            meta.labels.units: 'km',
+            meta.labels.name: 'ECI {:}-position'.format(v),
+            meta.labels.desc: 'Earth Centered Inertial {:}-position'.format(v),
```
Comment (JonathonMSmith) [inline pysatMissions/instruments/missions_skyfield.py:257]: Do we want to explicitly name the reference frame here, or keep this as the type of reference frame?

Comment (jklenzing) [reply pysatMissions/instruments/missions_skyfield.py:258]: Added to notes

File: pysatMissions/instruments/missions_skyfield.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    skytimes = skyapi.load.timescale().from_datetimes(index.tz_localize('UTC'))
+
+    esat = skyapi.EarthSatellite.from_satrec(satrec, skyapi.load.timescale())
+    # Generate object with info for all times
+    sat_obj = esat.at(skytimes)
+
+    position = sat_obj.position.km
+    velocity = sat_obj.velocity.km_per_s
+
+    # Add ECEF values to instrument.
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply pysatMissions/instruments/missions_skyfield.py:219]: added a comment

**Comment by jklenzing** [comment]
Pinging @rstoneback for review

**Comment by jklenzing** [comment]
pinging @rstoneback and @JonathonMSmith for review

**Comment by jklenzing** [comment]
make sure sgp4 meta is documented

**Review by rstoneback — APPROVED** [review]

**Pull request merged**